packages/ui/components/scheduler/components/Scheduler.tsx

```diff
@@ -0,0 +1,134 @@
+import { VeritcalLines } from "./verticalLines";
+
+export default function Scheduler(props: SchedulerComponentProps) {
+  const container = useRef<HTMLDivElement | null>(null);
+  const containerNav = useRef<HTMLDivElement | null>(null);
+  const containerOffset = useRef<HTMLDivElement | null>(null);
+  const initalState = useSchedulerStore((state) => state.initState);
+
+  const { startDate, endDate, startHour, endHour, usersCellsStopsPerHour } = useSchedulerStore((state) => ({
+    startDate: state.startDate,
+    endDate: state.endDate,
+    startHour: state.startHour || 0,
+    endHour: state.endHour || 23,
+    usersCellsStopsPerHour: state.gridCellsPerHour || 4,
+  }));
+
+  const days = useMemo(() => getDaysBetweenDates(startDate, endDate), [startDate, endDate]);
+
+  const hours = useMemo(() => getHoursToDisplay(startHour || 0, endHour || 23), [startHour, endHour]);
+
+  const numberOfGridStopsPerDay = hours.length * usersCellsStopsPerHour;
+
+  // Initalise State on inital mount
+  useEffect(() => {
+    initalState(props);
+  }, [props, initalState]);
+
+  return (
+    <MobileNotSupported>
+      <div
+        className="scheduler-wrapper flex h-full w-full flex-col overflow-y-scroll"
+        style={
+          { "--one-minute-height": `calc(1.75rem/(60/${usersCellsStopsPerHour}))` } as React.CSSProperties // This can't live in the css file because it's a dynamic value and css variable gets super
+        }>
+        <SchedulerHeading />
+        <div ref={container} className="relative isolate flex flex-auto  flex-col bg-white">
+          <div
+            style={{ width: "165%" }}
+            className="flex max-w-full flex-none flex-col sm:max-w-none md:max-w-full">
+            <DateValues containerNavRef={containerNav} days={days} />
+            {/* TODO: Implement this at a later date. */}
+            {/* <CurrentTime
+            containerNavRef={containerNav}
+            containerOffsetRef={containerOffset}
+            containerRef={container}
+          /> */}
+            <div className="flex flex-auto">
+              <div className="sticky left-0 z-10 w-14 flex-none bg-white ring-1 ring-gray-100" />
+              <div className="grid flex-auto grid-cols-1 grid-rows-1 ">
+                <HorizontalLines
+                  hours={hours}
+                  numberOfGridStopsPerCell={usersCellsStopsPerHour}
+                  containerOffsetRef={containerOffset}
+                />
+                <VeritcalLines days={days} />
+
+                {/* Empty Cells */}
+                <SchedulerColumns
+                  zIndex={50}
+                  offsetHeight={containerOffset.current?.offsetHeight}
+                  gridStopsPerDay={numberOfGridStopsPerDay}>
+                  <>
+                    {[...Array(days.length)].map((_, i) => (
+                      <li
+                        key={i}
+                        style={{
+                          gridRow: `2 / span ${numberOfGridStopsPerDay}`,
```

Comment:
we use `2` here due to the fact the first row is reserved for all day events (Not yet implemented)

Then we span all of the grid stops in the day